src/core/core.controller.js

```diff
@@ -104,15 +96,19 @@ function initConfig(config) {
 
 	// Do NOT use mergeConfig for the data object because this method merges arrays
 	// and so would change references to labels and datasets, preventing data updates.
-	var data = config.data = config.data || {};
+	const data = config.data = config.data || {};
 	data.datasets = data.datasets || [];
 	data.labels = data.labels || [];
 
+	const scaleConfig = mergeScaleConfig(config, config.options);
```

Comment:
I think it'd be nice to put this line and `config.options.scales = scaleConfig` inside `mergeConfig`. Then you don't need to duplicate them across `initConfig` and `updateConfig`

Reply:
`mergeConfig` gets only the `options` part, so no access to dataset types.
Lets do the fine tuning in another PR later?

Reply:
Sounds fine to me